database/database.go

```diff
@@ -301,6 +360,87 @@ func (db *Database) Close() {
  * Private
  **********************************************************************************************************************/
 
+func (db *Database) getDataFromQuery(query string, queryParams []any, result ...interface{}) error {
+	stmt, err := db.sql.Prepare(query)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+	defer stmt.Close()
+
+	if err = stmt.QueryRow(queryParams...).Scan(result...); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return errNotExist
+		}
+
+		return aoserrors.Wrap(err)
+	}
+
+	return nil
+}
+
+func (db *Database) getDownloadFromQuery(
+	query string, args ...interface{},
+) (downloadInfos []downloader.DownloadInfo, err error) {
+	rows, err := db.sql.Query(query, args...)
+	if err != nil {
+		return downloadInfos, aoserrors.Wrap(err)
+	}
+	defer rows.Close()
+
+	if rows.Err() != nil {
+		return nil, aoserrors.Wrap(rows.Err())
+	}
+
+	for rows.Next() {
+		var downloadInfo downloader.DownloadInfo
+
+		if err = rows.Scan(
+			&downloadInfo.Path, &downloadInfo.TargetType,
+			&downloadInfo.InterruptReason, &downloadInfo.Released); err != nil {
+			return nil, aoserrors.Wrap(err)
+		}
+
+		downloadInfos = append(downloadInfos, downloadInfo)
+	}
+
+	return downloadInfos, nil
+}
+
+func (db *Database) executeQuery(query string, args ...interface{}) error {
+	stmt, err := db.sql.Prepare(query)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+	defer stmt.Close()
+
+	result, err := stmt.Exec(args...)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	count, err := result.RowsAffected()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	if count == 0 {
+		return aoserrors.Wrap(errNotExist)
```

Comment:
I guess internal error should not be wrapped.

Reply:
in this case linter shows an error